lib/zendesk_apps_support/validations/svg.rb

```diff
@@ -4,33 +4,38 @@
 module ZendeskAppsSupport
   module Validations
     module Svg
-      @strip_declaration = Loofah::Scrubber.new do |node|
-        node.remove if node.name == 'xml' && node.children.empty?
-      end
-
-      @empty_malformed_markup = Loofah::Scrubber.new do |node|
-        node.next.remove while node.name == 'svg' && node.next
-      end
+      # whitelist elements and attributes used in Zendesk Garden assets
+      Loofah::HTML5::WhiteList::ALLOWED_ELEMENTS_WITH_LIBXML2.add 'symbol'
+      Loofah::HTML5::WhiteList::ACCEPTABLE_CSS_PROPERTIES.add 'position'
 
       # CRUFT: ignore a (very specific) style attribute which Loofah would otherwise scrub.
       # This attribute is deprecated (https://www.w3.org/TR/filter-effects/#AccessBackgroundImage)
       # but is included in many of the test apps used in fixtures for tests in ZAM, ZAT etc.
-      @remove_enable_background = Loofah::Scrubber.new do |node|
-        match_pattern = Regexp.new("enable-background:.*?(\;|\z)")
+      Loofah::HTML5::WhiteList::ACCEPTABLE_CSS_PROPERTIES.add 'enable-background'
+
+      @strip_declaration = Loofah::Scrubber.new do |node|
+        node.remove if node.name == 'xml' && node.children.empty?
+      end
+
+      @strip_spaces_between_css_attrs = Loofah::Scrubber.new do |node|
+        match_pattern = Regexp.new(/\;\s+/)
         if node.name == 'svg' && node['style']
-          node['style'] = node['style'].gsub(match_pattern, '')
-          node.attributes['style'].remove if node['style'].empty?
+          node['style'] = node['style'].gsub(match_pattern, ';')
         end
       end
 
+      @empty_malformed_markup = Loofah::Scrubber.new do |node|
+        node.next.remove while node.name == 'svg' && node.next
```

Comment:
Is this because we can only support one svg node?

Reply:
Yep, sort of. I think it was also so we didn't have subsequent tags following/outside the main <svg>. I do remember adding this to make the tests at https://github.com/zendesk/zendesk_apps_support/blob/cd87351ed0bff97347c655e59a3e512b5ba572b6/spec/validations/svg_spec.rb#L120 make sense